src/core/compile/sign/mod.rs

```diff
@@ -266,11 +280,30 @@ fn build_signature(
     let Located(loc, obj) = def;
 
     match obj {
-        ca::TopLevelStatementObj::ClassDef { body, bases, .. } => {
+        ca::TopLevelStatementObj::ClassDef {
+            body,
+            bases,
+            decorator_list,
+            ..
+        } => {
             let mut is_account = false;
             let mut is_enum = false;
             let mut is_event = false;
             let mut bases_ = vec![];
+            let mut is_dataclass = false;
+
+            for Located(loc, decorator) in decorator_list.into_iter() {
+                match decorator {
+                    ca::ExpressionObj::Id(d) if d == "dataclass" => {
```

Comment:
@mcintyre94 right here

Reply:
That'd definitely be really nice!

For now I've improved the error message when it's parsed as an `Attribute` which `@seahorse.prelude.dataclass` does:

```
$ ./target/debug/seahorse compile examples/event.py
Error: dataclass is not a supported class decorator.
11 | @seahorse.prelude.dataclass
                      ^
Hint: Only dataclass is currently supported for classes. Imported paths like "seahorse.prelude.dataclass" are not currently supported.
```

It's not perfect because the name ends up being `dataclass`, but I added a note specifically for imported paths and I think that + the error location info are good enough for now